### ISSUE TITLE
Issue during Playlist URI instantiation when there's a fourth part present

### ISSUE THREAD

**Issue by duberton** [adamint/spotify-web-api-kotlin#248]
**Describe the bug**

During the deserializing phase of a PlayHistoryContext, in case there's an URI that contains 4 parts like `spotify:playlist:4SIcFp6g2h85IX7jgIsuRL:recommended`, it ends up breaking the safe initialization of a PlaylistUri and throws a `SpotifyUriException`.

**Code mentions:**

In the `SpotifyUris.init block`, there's a function that removes the 4th part and then during `SpotifyUris:110` the safe initialization comparison made between the processed uri (without `:recommended`) and the original one (with `:recommended`) returns false, thus, the exception.

Dependency def: `com.adamratzman:spotify-api-kotlin-core:3.5.03`

**Comment by adamint** [comment]
I can’t find any reference in the Spotify docs to URIs with four parts. Looking more into this

**Comment by adamint** [comment]
I guess this is also a good time to start adding PlayerApi tests. I was holding off because they're beta, but that's not changing anytime soon

**Comment by adamint** [comment]
I'm unable to reproduce this @duberton

**Comment by duberton** [comment]
Sure. I'll try to put them here soon.

**Comment by adamint** [comment]
Thank you!

**Comment by duberton** [comment]
To be able to reproduce it you need to listen to a song that is being displayed at the 'recommended' section of the app (at the very bottom of playlist) and then perform a call to the `me/player/recently-played` endpoint.

I just did this and got the following json: 

```
{
  "items" : [ {
    "track" : {
      "album" : {
        "album_type" : "single",
        "artists" : [ {
          "external_urls" : {
            "spotify" : "https://open.spotify.com/artist/3h1aCZ3gZ4zIWxnsxcBrPD"
          },
          "href" : "https://api.spotify.com/v1/artists/3h1aCZ3gZ4zIWxnsxcBrPD",
          "id" : "3h1aCZ3gZ4zIWxnsxcBrPD",
          "name" : "Nico de Andrea",
          "type" : "artist",
          "uri" : "spotify:artist:3h1aCZ3gZ4zIWxnsxcBrPD"
        } ],
        "available_markets" : [ "AD", "AE", "AL", "AR", "AT", "AU", "BA", "BE", "BG", "BH", "BO", "BR", "BY", "CA", "CH", "CL", "CO", "CR", "CY", "CZ", "DE", "DK", "DO", "DZ", "EC", "EE", "EG", "ES", "FI", "FR", "GB", "GR", "GT", "HK", "HN", "HR", "HU", "ID", "IE", "IL", "IN", "IS", "IT", "JO", "JP", "KW", "KZ", "LB", "LI", "LT", "LU", "LV", "MA", "MC", "MD", "ME", "MK", "MT", "MX", "MY", "NI", "NL", "NO", "NZ", "OM", "PA", "PE", "PH", "PL", "PS", "PT", "PY", "QA", "RO", "RS", "RU", "SA", "SE", "SG", "SI", "SK", "SV", "TH", "TN", "TR", "TW", "UA", "US", "UY", "VN", "XK", "ZA" ],
        "external_urls" : {
          "spotify" : "https://open.spotify.com/album/4Pu0prIo3nGtMvwuSAXBbg"
        },
        "href" : "https://api.spotify.com/v1/albums/4Pu0prIo3nGtMvwuSAXBbg",
        "id" : "4Pu0prIo3nGtMvwuSAXBbg",
        "images" : [ {
          "height" : 640,
          "url" : "https://i.scdn.co/image/ab67616d0000b273c1e9b4bbf142b6e279ed5240",
          "width" : 640
        }, {
          "height" : 300,
          "url" : "https://i.scdn.co/image/ab67616d00001e02c1e9b4bbf142b6e279ed5240",
          "width" : 300
        }, {
          "height" : 64,
          "url" : "https://i.scdn.co/image/ab67616d00004851c1e9b4bbf142b6e279ed5240",
          "width" : 64
        } ],
        "name" : "The Shape",
        "release_date" : "2018-04-13",
        "release_date_precision" : "day",
        "total_tracks" : 1,
        "type" : "album",
        "uri" : "spotify:album:4Pu0prIo3nGtMvwuSAXBbg"
      },
      "artists" : [ {
        "external_urls" : {
          "spotify" : "https://open.spotify.com/artist/3h1aCZ3gZ4zIWxnsxcBrPD"
        },
        "href" : "https://api.spotify.com/v1/artists/3h1aCZ3gZ4zIWxnsxcBrPD",
        "id" : "3h1aCZ3gZ4zIWxnsxcBrPD",
        "name" : "Nico de Andrea",
        "type" : "artist",
        "uri" : "spotify:artist:3h1aCZ3gZ4zIWxnsxcBrPD"
      } ],
      "available_markets" : [ "AD", "AE", "AL", "AR", "AT", "AU", "BA", "BE", "BG", "BH", "BO", "BR", "BY", "CA", "CH", "CL", "CO", "CR", "CY", "CZ", "DE", "DK", "DO", "DZ", "EC", "EE", "EG", "ES", "FI", "FR", "GB", "GR", "GT", "HK", "HN", "HR", "HU", "ID", "IE", "IL", "IN", "IS", "IT", "JO", "JP", "KW", "KZ", "LB", "LI", "LT", "LU", "LV", "MA", "MC", "MD", "ME", "MK", "MT", "MX", "MY", "NI", "NL", "NO", "NZ", "OM", "PA", "PE", "PH", "PL", "PS", "PT", "PY", "QA", "RO", "RS", "RU", "SA", "SE", "SG", "SI", "SK", "SV", "TH", "TN", "TR", "TW", "UA", "US", "UY", "VN", "XK", "ZA" ],
      "disc_number" : 1,
      "duration_ms" : 174391,
      "explicit" : false,
      "external_ids" : {
        "isrc" : "NLZ541701244"
      },
      "external_urls" : {
        "spotify" : "https://open.spotify.com/track/5vA6wWjzDxFFIWsYh04jCu"
      },
      "href" : "https://api.spotify.com/v1/tracks/5vA6wWjzDxFFIWsYh04jCu",
      "id" : "5vA6wWjzDxFFIWsYh04jCu",
      "is_local" : false,
      "name" : "The Shape",
      "popularity" : 35,
      "preview_url" : "https://p.scdn.co/mp3-preview/69ba80512a5fee490b056d233649cc9fb92ad3b9?cid=ea55009abf82452189c8843e4d94ce74",
      "track_number" : 1,
      "type" : "track",
      "uri" : "spotify:track:5vA6wWjzDxFFIWsYh04jCu"
    },
    "played_at" : "2021-01-28T10:49:56.587Z",
    "context" : {
      "external_urls" : {
        "spotify" : "https://open.spotify.com/playlist/4SIcFp6g2h85IX7jgIsuRL/recommended"
      },
      "href" : "https://api.spotify.com/v1/playlists/4SIcFp6g2h85IX7jgIsuRL:recommended",
      "type" : "playlist",
      "uri" : "spotify:playlist:4SIcFp6g2h85IX7jgIsuRL:recommended"
    }
  } ],
  "next" : "https://api.spotify.com/v1/me/player/recently-played?after=1611830996587&limit=30",
  "cursors" : {
    "after" : "1611830996587",
    "before" : "1611830996587"
  },
  "limit" : 30,
  "href" : "https://api.spotify.com/v1/me/player/recently-played?after=1611797497110&limit=30"
}
```

**Comment by adamint** [comment]
I'm still not able to reproduce this, even after following what you said. 
I played this track:
![image](https://user-images.githubusercontent.com/20359921/106152547-8ebb1980-614b-11eb-8674-b5d57de8569d.png)
Then performed the following request after finishing:
![image](https://user-images.githubusercontent.com/20359921/106152610-a2668000-614b-11eb-854c-e6be161eae8d.png)
But context is null.

**Comment by duberton** [comment]
Yeah, it won't reproduce unless the context is returned :(

**Comment by adamint** [comment]
Not sure why I'm not receiving context

**Comment by adamint** [comment]
Here's what I can do:
I'll remove ":recommended" suffix from the SpotifyUri input. This will be a temporary workaround until I can figure out why :recommended is showing up in the first place (there's no documentation about it!)

**Comment by duberton** [comment]
Alright, thanks Adam.

**Comment by adamint** [comment]
@duberton Sorry this took so much longer to release, should be available in `3.5.05` (one hour)

**Comment by duberton** [comment]
Awesome. Thanks a lot Adam!
